checker/src/main/java/org/checkerframework/checker/initialization/InitializationChecker.java

```diff
@@ -4,37 +4,112 @@
+import org.checkerframework.checker.initialization.qual.HoldsForDefaultValues;
+import org.checkerframework.checker.initialization.qual.Initialized;
+import org.checkerframework.checker.nullness.NullnessChecker;
+import org.checkerframework.checker.nullness.NullnessNoInitAnnotatedTypeFactory;
+import org.checkerframework.checker.nullness.NullnessNoInitSubchecker;
+import org.checkerframework.checker.nullness.qual.EnsuresNonNull;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
 import org.checkerframework.common.basetype.BaseTypeChecker;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.NavigableSet;
+import java.util.Set;
 
 /**
  * Tracks whether a value is initialized (all its fields are set), and checks that values are
  * initialized before being used. Implements the freedom-before-commitment scheme for
  * initialization, augmented by type frames.
  *
+ * <p>Because there is a cyclic dependency between this type system and the target type system,
+ * using this checker is more complex than for others. Specifically, the target checker must:
+ *
+ * <ol>
+ *   <li>Use a subclass of this checker as its parent checker. This is necessary because this
+ *       checker is dependent on the target checker to know which fields should be checked for
+ *       initialization, and when such a field is initialized: A field is checked for initialization
+ *       if its declared type is not the top type and does not have the meta-annotation {@link
+ *       HoldsForDefaultValues} (e.g., {@link NonNull}). Such a field becomes initialized as soon as
+ *       its refined type agrees with its declared type (which can happen either by assigning the
+ *       field or by a contract annotation like {@link EnsuresNonNull}). You can look at the {@link
+ *       NullnessChecker} for an example: The {@link NullnessChecker} is a subclass of this checker
+ *       and uses the {@link NullnessNoInitSubchecker} as the target checker; thus, the {@link
+ *       NullnessNoInitSubchecker} actually checks {@link NonNull} and related qualifiers, while the
+ *       NullnessChecker checks {@link Initialized} and related qualifers.
+ *   <li>Use the {@link InitializationFieldAccessSubchecker} as a subchecker and add its {@link
+ *       CommitmentFieldAccessTreeAnnotator} as a tree annotator. This is necessary to give possibly
+ *       uninitialized fields the top type of the target hierarchy (e.g., {@link Nullable}),
+ *       ensuring that all fields are initialized before being used. This needs to be a separate
+ *       checker because the target checker cannot access any type information from its parent,
+ *       which is only initialized after all subcheckers have finished.
+ *   <li>Override all necessary methods in the target checker's type factory to take the type
+ *       information from the InitializationDeclarationChecker into account. You can look at {@link
```

Comment:
This looks like the only occurrence of `InitializationDeclarationChecker`. What class is it referring to?